pyo3cls/src/py_class.rs

```diff
@@ -99,7 +99,7 @@ fn impl_class(cls: &syn::Ident, base: &syn::Ident,
             }
             impl std::convert::AsRef<PyObjectRef> for #cls {
                 fn as_ref(&self) -> &_pyo3::PyObjectRef {
-                    unsafe{std::mem::transmute(self.as_ptr())}
+                    unsafe{&*(self.as_ptr() as *const pyo3::PyObjectRef)}
```

Comment:
looks interesting, but it wouldn't help. pyo3 memory leaks was logical and specific to python ref counting